spec/granite/fields/casting_spec.cr

```diff
@@ -1,9 +1,22 @@
+require "../../spec_helper"
+
 describe "#casting_to_fields" do
   it "casts string to int" do
     model = Review.new({"downvotes" => "32"})
     model.downvotes.should eq 32
   end
 
+  it "casts time with timezone" do
+    timezone = Granite.settings.default_timezone
+    Granite.settings.default_timezone = Time::Location.load("Asia/Shanghai")
+    created_at = Time.parse("2018-12-12 00:00:00 +00:00", "%F %T %:z", Time::Location::UTC)
+    model = Review.new({"created_at" => created_at})
+    model.created_at.should eq Time.parse("2018-12-11 16:00:00 +08:00", "%F %T %:z", Granite.settings.default_timezone)
+
+    # Restore previous timezone
+    Granite.settings.default_timezone = timezone
```

Comment:
Instead of restoring here, should we reset this in a `before_each`?  Not sure this will ever be an issue but might make it a little more stable if this failed to restore it.

Reply:
I force push commit and add `before_each`